### PR TITLE
feat: (Platform) add Object Marker documentation and platform library changes

### PR DESCRIPTION
**Pull request by Lokanathan-k**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/2601

#### Please provide a brief summary of this pull request.

This indicates the technical status of an object.
It is enabled for the dynamic page layout, snapping header, object page header, upload collection, and object list item.
The technical status can be represented as and icon, with an icon and text, or as text only, depending on the screen size.
Currently, the following technical statuses can be visualized by the object marker:

Flag
Favorite
Editing Status (Draft, Locked, Unsaved Changes)
The editing status is part of the draft handling concept.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/object-marker/object-marker.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ObjectMarkerComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (sKudum) [inline libs/platform/src/lib/components/object-marker/object-marker.component.spec.ts:24]: having style class check will help to cover a few tests senarios

**Review by sKudum — CHANGES_REQUESTED** [review]
It does not have breaking changes, we can remove the selected option PR description.

**Review by Betrozov — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Comment by sKudum** [comment]
Hi Lok, 
I am approving it, please address API issues before proceeding further that will help.

**Review by sKudum — APPROVED** [review]

File: libs/platform/src/lib/components/object-marker/object-marker.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    template: `<fdp-object-marker [glyph]="glyph" [clickable]="clickable">Object marker </fdp-object-marker>`
+})
+class TestPlatformPlatformObjectMarkerComponent {
+    @ViewChild(PlatformObjectMarkerComponent, { static: true }) component: PlatformObjectMarkerComponent;
+    public glyph: string;
+    public clickable: boolean;
+    constructor() {}
+}
+
+fdescribe('PlatformObjectMarkerComponent', () => {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/object-marker/object-marker.component.spec.ts:18]: Please remove the focus

**Review by InnaAtanasova — APPROVED** [review]
Please remove the focus from the test. The rest looks good.

**Comment by Lokanathan-k** [comment]
@InnaAtanasova @sKudum  done with the code change will be merging the code.

**Pull request merged**